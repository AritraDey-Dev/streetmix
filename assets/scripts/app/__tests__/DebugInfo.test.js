/* eslint-env jest */
import React from 'react'
import { cleanup, fireEvent } from '@testing-library/react'
import { renderWithRedux } from '../../../../test/helpers/render'
import DebugInfo from '../DebugInfo'

describe('DebugInfo', () => {
  const initialState = {
    settings: {},
    street: {},
    flags: {},
    undo: {},
    user: {}
  }

  it('renders', () => {
    const wrapper = renderWithRedux(<DebugInfo />, { initialState })
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  // fireEvent isn't working.
  it.skip('is visible when opened with keyboard shortcut', () => {
    const wrapper = renderWithRedux(<DebugInfo />, { initialState })
    fireEvent.keyDown(window, { key: 'D', code: 68, shiftKey: true })
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
