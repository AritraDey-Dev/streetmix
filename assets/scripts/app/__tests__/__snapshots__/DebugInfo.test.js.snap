// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebugInfo renders 1`] = `
<DocumentFragment>
  <div
    class="debug-info"
  >
    <textarea
      readonly=""
      wrap="off"
    />
  </div>
</DocumentFragment>
`;
